lib/usd/pxrUsdPreviewSurface/usdPreviewSurface.cpp

```diff
@@ -388,10 +371,15 @@ PxrMayaUsdPreviewSurface::compute(const MPlug& plug, MDataBlock& dataBlock)
 
     // XXX: For now, we simply propagate diffuseColor to outColor and
     // opacity to outTransparency.
-
+    MFnDependencyNode depNodeFn(thisMObject());
+    MObject           outColorAttr
+        = depNodeFn.attribute(PxrMayaUsdPreviewSurfaceTokens->OutColorAttrName.GetText());
+    MObject outTransparencyAttr
+        = depNodeFn.attribute(PxrMayaUsdPreviewSurfaceTokens->OutTransparencyAttrName.GetText());
```

Comment:
I had thought there would be a performance penalty with frequently looking up attributes by name like this, which is probably what leads to the pattern strewn around the repo of attributes being declared as static `MObject`s. Is that not the case?

Reply:
We end up with 2 sets of non-interchangeable MObjects and this causes very subtle bugs on export depending on the registration order of the Pixar and Autodesk plugins. The search by name ensures the export will succeed, but has a non-zero performance cost. If you think this will negatively affect the performance of the export and the evaluation, I can create a static cache for the 2 sets of MObjects.

Reply:
Ok, cool! Nope, no real concern with performance in this particular case. I was just curious what motivated the change. Makes sense, thanks!